lib/torrent.js

```diff
@@ -97,7 +97,7 @@ function Torrent (torrentId, opts) {
 Object.defineProperty(Torrent.prototype, 'timeRemaining', {
   get: function () {
     if (this.done) return 0
-    if (this.swarm.downloadSpeed() === 0) return Infinity
+    if (this.swarm === undefined || this.swarm.downloadSpeed() === 0) return Infinity
```

Comment:
This would also fail if swarm is `null`. It's better to just to do:

```js
if (!this.swarm || this.swarm.downloadSpeed() === 0) return Infinity
```